.github/workflows/functional-test.yaml

```diff
@@ -609,7 +609,7 @@ jobs:
         uses: actions/upload-artifact@v4
         if: always()
         with:
-          name: recipes-pod-logs
+          name: ${{ matrix.name }}_recipes-pod-logs
```

Comment:
From v4, upload artifact doesn't support append operation. We should have the explicit name for each test.